test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -79,167 +87,330 @@ func (s *snapshottableTestSuite) SkipRedundantSuite(driver TestDriver, pattern t
+func WaitForSnapshotReady(c dynamic.Interface, ns string, snapshotName string, poll, timeout time.Duration) error {
 	framework.Logf("Waiting up to %v for VolumeSnapshot %s to become ready", timeout, snapshotName)
-	for start := time.Now(); time.Since(start) < timeout; time.Sleep(Poll) {
+
+	if successful := WaitUntil(poll, timeout, func() bool {
 		snapshot, err := c.Resource(SnapshotGVR).Namespace(ns).Get(context.TODO(), snapshotName, metav1.GetOptions{})
 		if err != nil {
-			framework.Logf("Failed to get claim %q, retrying in %v. Error: %v", snapshotName, Poll, err)
-			continue
-		} else {
-			status := snapshot.Object["status"]
-			if status == nil {
-				framework.Logf("VolumeSnapshot %s found but is not ready.", snapshotName)
-				continue
-			}
-			value := status.(map[string]interface{})
-			if value["readyToUse"] == true {
-				framework.Logf("VolumeSnapshot %s found and is ready after %v", snapshotName, time.Since(start))
-				return nil
-			} else if value["ready"] == true {
-				framework.Logf("VolumeSnapshot %s found and is ready after %v", snapshotName, time.Since(start))
-				return nil
-			} else {
-				framework.Logf("VolumeSnapshot %s found but is not ready.", snapshotName)
-			}
+			framework.Logf("Failed to get snapshot %q, retrying in %v. Error: %v", snapshotName, poll, err)
+			return false
 		}
+
+		status := snapshot.Object["status"]
+		if status == nil {
+			framework.Logf("VolumeSnapshot %s found but is not ready.", snapshotName)
+			return false
+		}
+		value := status.(map[string]interface{})
+		if value["readyToUse"] == true {
+			framework.Logf("VolumeSnapshot %s found and is ready", snapshotName)
+			return true
+		}
+
+		framework.Logf("VolumeSnapshot %s found but is not ready.", snapshotName)
+		return false
+	}); successful {
+		return nil
 	}
+
 	return fmt.Errorf("VolumeSnapshot %s is not ready within %v", snapshotName, timeout)
 }
+
+// WaitForGVRDeletion waits until an object has been deleted
+func WaitForGVRDeletion(c dynamic.Interface, gvr schema.GroupVersionResource, objectName string, poll, timeout time.Duration) error {
+	framework.Logf("Waiting up to %v for %s %s to be deleted", timeout, gvr.Resource, objectName)
+
+	if successful := WaitUntil(poll, timeout, func() bool {
+		_, err := c.Resource(gvr).Get(context.TODO(), objectName, metav1.GetOptions{})
+		if err != nil && apierrors.IsNotFound(err) {
+			framework.Logf("%s %v is not found and has been deleted", gvr.Resource, objectName)
+			return true
+		} else if err != nil {
+			framework.Logf("Get $s %v returned an error: %v", objectName, err.Error())
+		} else {
+			framework.Logf("%s %v has been found and is not deleted", gvr.Resource, objectName)
+		}
+
+		return false
+	}); successful {
+		return nil
+	}
+
+	return fmt.Errorf("%s %s is not deleted within %v", gvr.Resource, objectName, timeout)
+}
+
+// WaitUntil runs checkDone until a timeout is reached
+func WaitUntil(poll, timeout time.Duration, checkDone func() bool) bool {
```

Comment:
nit: This looks like a function which can be placed in a common utils

Reply:
Moved to utils :+1: